File: momentra/momentra/Shell/GroupActive/Living/Create/LivingQuickAddSheets.swift
import SwiftUI

/// Living Quick Add sheets — theme-accented; House Rule is a GAP (disabled CTA).
struct LivingGapQuickAddSheet: View {
    let theme: LivingActiveTheme
    let kind: LivingQuickAddKind
    var momentId: String? = nil
    var momentTypeCode: String? = nil
    var onClose: () -> Void
    var onSaved: () -> Void = {}

    private var accent: SheetAccent {
        SheetAccent(accent: theme.accent, accentEnd: theme.accentSolid, soft: theme.accentSoft)
    }

    var body: some View {
        if kind == .expense, let momentId {
            GroupExpenseSheet(
                momentId: momentId,
                isPresented: Binding(
                    get: { true },
                    set: { if !$0 { onClose() } }
                ),
                momentTypeCode: momentTypeCode,
                onSaved: {
                    onSaved()
                    onClose()
                }
            )
        } else {
            NativeSheetScaffold(
                title: kind.label,
                onClose: onClose,
                background: Color(hex: "#1C1A24")
            ) {
                ScrollView {
                    sheetBody
                        .padding(.horizontal, 20)
                        .padding(.top, 16)
                }
            }
            .presentationDetents([.large])
            .presentationDragIndicator(.visible)
        }
    }

    @ViewBuilder
    private var sheetBody: some View {
        switch kind {
        case .expense:
            EmptyView()
        case .contribution:
            WeddingContributionBody(momentId: momentId, onDismiss: onClose, onSaved: onSaved, accent: accent)
        case .poll:
            WeddingPollBody(momentId: momentId, onDismiss: onClose, onSaved: onSaved, accent: accent)
        case .update:
            WeddingUpdateBody(momentId: momentId, onDismiss: onClose, onSaved: onSaved, accent: accent)
        case .memory:
            WeddingMemoryBody(momentId: momentId, onDismiss: onClose, onSaved: onSaved, accent: accent)
        case .task:
            WeddingPlanningBody(momentId: momentId, onDismiss: onClose, onSaved: onSaved, accent: accent)
        case .resident:
            LivingResidentBody(momentId: momentId, theme: theme, onDismiss: onClose, onSaved: onSaved, accent: accent)
        case .asset:
            LivingSharedAssetBody(momentId: momentId, onDismiss: onClose, onSaved: onSaved, accent: accent)
        case .maintenance:
            LivingMaintenanceBody(momentId: momentId, onDismiss: onClose, onSaved: onSaved, accent: accent)
        case .rule:
            LivingHouseRuleBody(momentId: momentId, theme: theme, onDismiss: onClose, onSaved: onSaved, accent: accent)
        }
    }
}

// MARK: - Resident (live)

private struct LivingResidentBody: View {
    var momentId: String?
    let theme: LivingActiveTheme
    var onDismiss: () -> Void
    var onSaved: () -> Void
    var accent: SheetAccent

    @State private var name = ""
    @State private var contact = ""
    @State private var room = ""
    @State private var moveInDate = ""
    @State private var rentShare = ""
    @State private var role = ""
    @State private var sendInvitation = true
    @State private var addToExpenseSplit = true
    @State private var busy = false
    @State private var error: String?

    private let roomOptions = ["Unassigned", "Room 1", "Room 2", "Room 3", "Common area"]

    private var canSave: Bool {
        momentId != nil && !name.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty && !busy
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            SheetHeader(
                icon: "person.badge.plus",
                title: "Add Resident",
                subtitle: theme.participantSubtitle,
                accent: accent
            )

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Full Name")
                SheetField(value: $name, placeholder: "Full name", minHeight: 42)
            }

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Email or Phone")
                SheetField(value: $contact, placeholder: "email@example.com or +91…", minHeight: 42)
            }

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Room Assignment")
                Menu {
                    ForEach(roomOptions, id: \.self) { option in
                        Button(option) { room = option }
                    }
                } label: {
                    SheetField(
                        value: .constant(room.isEmpty ? "" : room),
                        placeholder: "Select room",
                        minHeight: 42,
                        trailing: {
                            AnyView(
                                Image(systemName: "chevron.down")
                                    .font(.system(size: 12))
                                    .foregroundStyle(Color(hex: "#9E9AA8"))
                            )
                        }
                    )
                }
            }

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Move-in Date")
                WeddingDatePickField(value: $moveInDate, placeholder: "Select date")
            }

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Monthly Rent Share")
                SheetField(
                    value: $rentShare,
                    placeholder: "0.00",
                    minHeight: 42,
                    leading: {
                        AnyView(
                            Text("₹")
                                .font(.plusJakarta(size: 18, weight: .bold))
                                .foregroundStyle(accent.accent)
                        )
                    },
                    keyboardType: .decimalPad
                )
                .onChange(of: rentShare) { _, new in
                    rentShare = new.filter { $0.isNumber || $0 == "." }
                }
            }

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Role")
                ChipRow(options: theme.participantRoles, selected: $role, accent: accent)
            }

            LivingToggleRow(
                title: "Send Invitation",
                subtitle: "Email/SMS joining link to this person",
                isOn: $sendInvitation,
                accent: accent.accent
            )
            LivingToggleRow(
                title: "Add to Expense Split",
                subtitle: "Auto-add to active shared bills",
                isOn: $addToExpenseSplit,
                accent: accent.accent
            )

            if let error {
                Text(error)
                    .font(.plusJakarta(size: 12))
                    .foregroundStyle(Color(hex: "#F87171"))
            }

            PrimaryCta(
                label: busy ? "Saving…" : (sendInvitation ? "Invite Resident" : "Add Resident"),
                enabled: canSave,
                accent: accent,
                loading: busy,
                lightLabel: true
            ) {
                Task { await save() }
            }
        }
        .onAppear {
            if role.isEmpty {
                role = theme.participantRoles.first ?? "Resident"
            }
        }
    }

    private func save() async {
        guard let momentId else { return }
        let trimmed = name.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return }
        // contact / room / moveInDate / rentShare / toggles: SCHEMA_GAP — UI parity only
        busy = true
        error = nil
        do {
            _ = try await APIClient.shared.addResident(
                momentId: momentId,
                name: trimmed,
                roleCode: role.isEmpty ? nil : role.uppercased().replacingOccurrences(of: " ", with: "_")
            )
            onSaved()
            onDismiss()
        } catch {
            self.error = error.localizedDescription
        }
        busy = false
    }
}

private struct LivingToggleRow: View {
    let title: String
    let subtitle: String
    @Binding var isOn: Bool
    var accent: Color

    var body: some View {
        HStack(alignment: .center) {
            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(.plusJakarta(size: 14, weight: .semibold))
                    .foregroundStyle(Color(hex: "#E5E0EE"))
                Text(subtitle)
                    .font(.plusJakarta(size: 11))
                    .foregroundStyle(Color(hex: "#9E9AA8"))
            }
            Spacer()
            Toggle("", isOn: $isOn)
                .labelsHidden()
                .tint(accent)
        }
    }
}

// MARK: - Shared Asset (live)

private struct LivingSharedAssetBody: View {
    var momentId: String?
    var onDismiss: () -> Void
    var onSaved: () -> Void
    var accent: SheetAccent

    @State private var title = ""
    @State private var assetType = ""
    @State private var condition = "GOOD"
    @State private var busy = false
    @State private var error: String?

    private let conditions = ["NEW", "GOOD", "FAIR", "POOR", "OUT_OF_SERVICE"]

    private var canSave: Bool {
        momentId != nil && !title.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty && !busy
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            SheetHeader(
                icon: "shippingbox.fill",
                title: "Add Shared Asset",
                subtitle: "Track something the household owns together",
                accent: accent
            )

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Asset Title")
                SheetField(value: $title, placeholder: "What is shared?", minHeight: 42)
            }

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Asset Type (optional)")
                SheetField(value: $assetType, placeholder: "Furniture, appliance, tool…", minHeight: 42)
            }

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Condition")
                ChipRow(options: conditions, selected: $condition, accent: accent)
            }

            if let error {
                Text(error)
                    .font(.plusJakarta(size: 12))
                    .foregroundStyle(Color(hex: "#F87171"))
            }

            PrimaryCta(
                label: busy ? "Saving…" : "Add Shared Asset",
                enabled: canSave,
                accent: accent,
                loading: busy,
                lightLabel: true
            ) {
                Task { await save() }
            }
        }
    }

    private func save() async {
        guard let momentId else { return }
        let trimmed = title.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return }
        busy = true
        error = nil
        do {
            let type = assetType.trimmingCharacters(in: .whitespacesAndNewlines)
            _ = try await APIClient.shared.createSharedAsset(
                momentId: momentId,
                title: trimmed,
                assetType: type.isEmpty ? nil : type,
                conditionCode: condition.isEmpty ? nil : condition
            )
            onSaved()
            onDismiss()
        } catch {
            self.error = error.localizedDescription
        }
        busy = false
    }
}

// MARK: - Maintenance (live)

private struct LivingMaintenanceBody: View {
    var momentId: String?
    var onDismiss: () -> Void
    var onSaved: () -> Void
    var accent: SheetAccent

    @State private var title = ""
    @State private var description = ""
    @State private var busy = false
    @State private var error: String?

    private var canSave: Bool {
        momentId != nil && !title.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty && !busy
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            SheetHeader(
                icon: "wrench.and.screwdriver.fill",
                title: "Log Maintenance",
                subtitle: "Record household upkeep or repairs",
                accent: accent
            )

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Title")
                SheetField(value: $title, placeholder: "What needs attention?", minHeight: 42)
            }

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Description")
                SheetField(value: $description, placeholder: "Details, vendor, notes…", singleLine: false, minHeight: 64)
            }

            if let error {
                Text(error)
                    .font(.plusJakarta(size: 12))
                    .foregroundStyle(Color(hex: "#F87171"))
            }

            PrimaryCta(
                label: busy ? "Saving…" : "Log Maintenance",
                enabled: canSave,
                accent: accent,
                loading: busy,
                lightLabel: true
            ) {
                Task { await save() }
            }
        }
    }

    private func save() async {
        guard let momentId else { return }
        let trimmed = title.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return }
        busy = true
        error = nil
        do {
            let desc = description.trimmingCharacters(in: .whitespacesAndNewlines)
            _ = try await APIClient.shared.createMaintenanceRecord(
                momentId: momentId,
                title: trimmed,
                description: desc.isEmpty ? nil : desc
            )
            onSaved()
            onDismiss()
        } catch {
            self.error = error.localizedDescription
        }
        busy = false
    }
}

// MARK: - House Rule

private struct LivingHouseRuleBody: View {
    var momentId: String?
    let theme: LivingActiveTheme
    var onDismiss: () -> Void
    var onSaved: () -> Void
    var accent: SheetAccent

    @State private var title = ""
    @State private var ruleText = ""
    @State private var busy = false
    @State private var error: String?

    private var canSave: Bool {
        momentId != nil
            && !title.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
            && !ruleText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
            && !busy
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            SheetHeader(
                icon: "list.bullet.rectangle",
                title: "House Rule",
                subtitle: "Set a shared expectation for the \(theme.typeLabel.lowercased())",
                accent: accent
            )

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Rule Title")
                SheetField(value: $title, placeholder: "Quiet hours, dishes, guests…", minHeight: 42)
            }

            VStack(alignment: .leading, spacing: 8) {
                FieldLabel(text: "Rule Details")
                SheetField(value: $ruleText, placeholder: "Describe the house rule…", singleLine: false, minHeight: 96)
            }

            if let error {
                Text(error)
                    .font(.plusJakarta(size: 12))
                    .foregroundStyle(Color(hex: "#F87171"))
            }

            PrimaryCta(
                label: busy ? "Saving…" : "Save House Rule",
                enabled: canSave,
                accent: accent,
                loading: busy,
                lightLabel: true
            ) {
                Task { await save() }
            }
        }
    }

    private func save() async {
        guard let momentId else { return }
        let trimmedTitle = title.trimmingCharacters(in: .whitespacesAndNewlines)
        let trimmedRule = ruleText.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmedTitle.isEmpty, !trimmedRule.isEmpty else { return }
        busy = true
        error = nil
        do {
            _ = try await APIClient.shared.createLivingRule(momentId: momentId, title: trimmedTitle, ruleText: trimmedRule)
            onSaved()
            onDismiss()
        } catch {
            self.error = error.localizedDescription
        }
        busy = false
    }
}

File: apk/app/src/main/java/com/example/momentra/ui/shell/group/living/create/LivingQuickAddSheets.kt
package com.example.momentra.ui.shell.group.living.create

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.momentra.R
import com.example.momentra.data.repository.GroupSliceRepository
import com.example.momentra.ui.shell.group.shared.GroupExpenseSheet
import com.example.momentra.ui.shell.group.shared.TripDatePickField
import com.example.momentra.ui.shell.group.wedding.create.ChipRow
import com.example.momentra.ui.shell.group.wedding.create.FieldLabel
import com.example.momentra.ui.shell.group.wedding.create.PrimaryCta
import com.example.momentra.ui.shell.group.wedding.create.SheetAccent
import com.example.momentra.ui.shell.group.wedding.create.SheetField
import com.example.momentra.ui.shell.group.wedding.create.SheetHeader
import com.example.momentra.ui.shell.group.wedding.create.WeddingContributionSheetBody
import com.example.momentra.ui.shell.group.wedding.create.WeddingMemorySheetBody
import com.example.momentra.ui.shell.group.wedding.create.WeddingPlanningSheetBody
import com.example.momentra.ui.shell.group.wedding.create.WeddingPollSheetBody
import com.example.momentra.ui.shell.group.wedding.create.WeddingUpdateSheetBody
import com.example.momentra.ui.theme.PlusJakartaSans
import kotlinx.coroutines.launch

private val EqSheet = Color(0xFF1C1A24)
private val EqHandle = Color(0xFF625E70)
private val EqMuted = Color(0xFF9E9AA8)

private fun LivingActiveTheme.toSheetAccent(): SheetAccent =
    SheetAccent(accent = accent, accentEnd = accentSolid, soft = accentSoft)

/** Shared Living Quick Add sheets — theme accent; House Rule is GAP. */
@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
@Composable
fun LivingGapQuickAddSheet(
    theme: LivingActiveTheme,
    kind: LivingQuickAddKind,
    visible: Boolean,
    momentId: String?,
    onDismiss: () -> Unit,
    onSaved: () -> Unit = {},
    momentTypeCode: String? = null,
    repository: GroupSliceRepository = remember { GroupSliceRepository() },
) {
    if (!visible) return

    if (kind == LivingQuickAddKind.EXPENSE && !momentId.isNullOrBlank()) {
        GroupExpenseSheet(
            momentId = momentId,
            visible = true,
            onDismiss = onDismiss,
            onSaved = {
                onSaved()
                onDismiss()
            },
            momentTypeCode = momentTypeCode,
            repository = repository,
        )
        return
    }

    val accent = theme.toSheetAccent()
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        containerColor = EqSheet,
        dragHandle = null,
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .navigationBarsPadding()
                .verticalScroll(rememberScrollState())
                .padding(horizontal = 20.dp)
                .padding(top = 12.dp, bottom = 28.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp),
        ) {
            Box(
                modifier = Modifier
                    .align(Alignment.CenterHorizontally)
                    .size(width = 48.dp, height = 4.dp)
                    .clip(RoundedCornerShape(2.dp))
                    .background(EqHandle),
            )
            when (kind) {
                LivingQuickAddKind.EXPENSE -> Unit
                LivingQuickAddKind.CONTRIBUTION ->
                    WeddingContributionSheetBody(momentId, repository, onDismiss, onSaved, accent)
                LivingQuickAddKind.POLL ->
                    WeddingPollSheetBody(momentId, repository, onDismiss, onSaved, accent)
                LivingQuickAddKind.MEMORY ->
                    WeddingMemorySheetBody(momentId, repository, onDismiss, onSaved, accent)
                LivingQuickAddKind.UPDATE ->
                    WeddingUpdateSheetBody(momentId, repository, onDismiss, onSaved, accent)
                LivingQuickAddKind.RESIDENT ->
                    LivingResidentSheetBody(theme, momentId, repository, onDismiss, onSaved, accent)
                LivingQuickAddKind.TASK ->
                    WeddingPlanningSheetBody(
                        momentId,
                        repository,
                        onDismiss,
                        onSaved,
                        accent,
                        momentTypeCode = momentTypeCode,
                    )
                LivingQuickAddKind.ASSET ->
                    LivingAssetSheetBody(momentId, repository, onDismiss, onSaved, accent)
                LivingQuickAddKind.MAINTENANCE ->
                    LivingMaintenanceSheetBody(momentId, repository, onDismiss, onSaved, accent)
                LivingQuickAddKind.RULE ->
                    LivingHouseRuleSheetBody(momentId, repository, onDismiss, onSaved, accent)
            }
        }
    }
}

@Composable
private fun LivingResidentSheetBody(
    theme: LivingActiveTheme,
    momentId: String?,
    repository: GroupSliceRepository,
    onDismiss: () -> Unit,
    onSaved: () -> Unit,
    accent: SheetAccent,
) {
    var name by remember { mutableStateOf("") }
    var contact by remember { mutableStateOf("") }
    var room by remember { mutableStateOf("") }
    var moveInDate by remember { mutableStateOf("") }
    var rentShare by remember { mutableStateOf("") }
    var role by remember { mutableStateOf(theme.participantRoles.first()) }
    var sendInvitation by remember { mutableStateOf(true) }
    var addToExpenseSplit by remember { mutableStateOf(true) }
    var roomMenuOpen by remember { mutableStateOf(false) }
    var submitting by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()
    val roomOptions = remember { listOf("Unassigned", "Room 1", "Room 2", "Room 3", "Common area") }

    SheetHeader(
        R.drawable.ic_qa_users,
        "Add Resident",
        theme.participantSubtitle,
        accent = accent,
    )
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Full Name")
        SheetField(name, { name = it }, "Full name", minHeight = 42)
    }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Email or Phone")
        SheetField(contact, { contact = it }, "email@example.com or +91…", minHeight = 42)
    }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Room Assignment")
        Box {
            SheetField(
                value = room,
                onValueChange = {},
                placeholder = "Select room",
                minHeight = 42,
                trailing = {
                    Icon(
                        painter = painterResource(R.drawable.ic_biz_create_chevron),
                        contentDescription = null,
                        tint = Color(0xFF9E9AA8),
                        modifier = Modifier.size(16.dp),
                    )
                },
            )
            // Make whole field open the menu
            Box(
                modifier = Modifier
                    .matchParentSize()
                    .clickable { roomMenuOpen = true },
            )
            DropdownMenu(
                expanded = roomMenuOpen,
                onDismissRequest = { roomMenuOpen = false },
            ) {
                roomOptions.forEach { option ->
                    DropdownMenuItem(
                        text = { Text(option, fontFamily = PlusJakartaSans) },
                        onClick = {
                            room = option
                            roomMenuOpen = false
                        },
                    )
                }
            }
        }
    }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Move-in Date")
        TripDatePickField(
            value = moveInDate,
            onValueChange = { moveInDate = it },
            placeholder = "Select date",
        )
    }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Monthly Rent Share")
        SheetField(
            rentShare,
            { rentShare = it.filter { c -> c.isDigit() || c == '.' } },
            "0.00",
            minHeight = 42,
            leading = {
                Text("₹", color = accent.accent, fontSize = 18.sp, fontWeight = FontWeight.Bold, fontFamily = PlusJakartaSans)
            },
            keyboardType = KeyboardType.Decimal,
        )
    }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Role")
        ChipRow(theme.participantRoles, role, accent) { role = it }
    }
    LivingToggleRow(
        title = "Send Invitation",
        subtitle = "Email/SMS joining link to this person",
        checked = sendInvitation,
        accent = accent.accent,
        onChange = { sendInvitation = it },
    )
    LivingToggleRow(
        title = "Add to Expense Split",
        subtitle = "Auto-add to active shared bills",
        checked = addToExpenseSplit,
        accent = accent.accent,
        onChange = { addToExpenseSplit = it },
    )
    error?.let { Text(it, color = Color(0xFFF87171), fontSize = 12.sp, fontFamily = PlusJakartaSans) }
    PrimaryCta(
        label = if (sendInvitation) "Invite Resident" else "Add Resident",
        enabled = !momentId.isNullOrBlank() && name.isNotBlank() && !submitting,
        loading = submitting,
        accent = accent,
        lightLabel = true,
        onClick = {
            val id = momentId ?: return@PrimaryCta
            // contact / room / moveInDate / rentShare / toggles: SCHEMA_GAP — UI parity only until API frozen
            scope.launch {
                submitting = true
                error = null
                repository.addResident(id, name.trim(), role).fold(
                    onSuccess = {
                        submitting = false
                        onSaved()
                        onDismiss()
                    },
                    onFailure = {
                        submitting = false
                        error = it.message
                    },
                )
            }
        },
    )
}

@Composable
private fun LivingToggleRow(
    title: String,
    subtitle: String,
    checked: Boolean,
    accent: Color,
    onChange: (Boolean) -> Unit,
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(title, color = Color(0xFFE5E0EE), fontSize = 14.sp, fontWeight = FontWeight.SemiBold, fontFamily = PlusJakartaSans)
            Text(subtitle, color = EqMuted, fontSize = 11.sp, fontFamily = PlusJakartaSans)
        }
        Switch(
            checked = checked,
            onCheckedChange = onChange,
            colors = SwitchDefaults.colors(checkedTrackColor = accent, checkedThumbColor = Color.White),
        )
    }
}

@Composable
private fun LivingAssetSheetBody(
    momentId: String?,
    repository: GroupSliceRepository,
    onDismiss: () -> Unit,
    onSaved: () -> Unit,
    accent: SheetAccent,
) {
    var title by remember { mutableStateOf("") }
    var assetType by remember { mutableStateOf("") }
    var condition by remember { mutableStateOf("GOOD") }
    var submitting by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()

    SheetHeader(
        R.drawable.ic_pulse_cart,
        "Add Shared Asset",
        "Track something the household shares",
        accent = accent,
    )
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Title")
        SheetField(title, { title = it }, "Asset name", minHeight = 42)
    }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Type (optional)")
        SheetField(assetType, { assetType = it }, "Appliance, furniture…", minHeight = 42)
    }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Condition")
        ChipRow(listOf("NEW", "GOOD", "FAIR", "POOR", "OUT_OF_SERVICE"), condition, accent) { condition = it }
    }
    error?.let { Text(it, color = Color(0xFFF87171), fontSize = 12.sp, fontFamily = PlusJakartaSans) }
    PrimaryCta(
        label = "Save Asset",
        enabled = !momentId.isNullOrBlank() && title.isNotBlank() && !submitting,
        loading = submitting,
        accent = accent,
        lightLabel = true,
        onClick = {
            val id = momentId ?: return@PrimaryCta
            scope.launch {
                submitting = true
                error = null
                repository.createSharedAsset(
                    id,
                    title.trim(),
                    assetType.trim().ifBlank { null },
                    condition,
                ).fold(
                    onSuccess = {
                        submitting = false
                        onSaved()
                        onDismiss()
                    },
                    onFailure = {
                        submitting = false
                        error = it.message
                    },
                )
            }
        },
    )
}

@Composable
private fun LivingMaintenanceSheetBody(
    momentId: String?,
    repository: GroupSliceRepository,
    onDismiss: () -> Unit,
    onSaved: () -> Unit,
    accent: SheetAccent,
) {
    var title by remember { mutableStateOf("") }
    var description by remember { mutableStateOf("") }
    var submitting by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()

    SheetHeader(
        R.drawable.ic_qa_sliders,
        "Add Maintenance",
        "Log a repair or upkeep item",
        accent = accent,
    )
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Title")
        SheetField(title, { title = it }, "What needs attention", minHeight = 42)
    }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Description (optional)")
        SheetField(description, { description = it }, "Details", minHeight = 42)
    }
    error?.let { Text(it, color = Color(0xFFF87171), fontSize = 12.sp, fontFamily = PlusJakartaSans) }
    PrimaryCta(
        label = "Save Maintenance",
        enabled = !momentId.isNullOrBlank() && title.isNotBlank() && !submitting,
        loading = submitting,
        accent = accent,
        lightLabel = true,
        onClick = {
            val id = momentId ?: return@PrimaryCta
            scope.launch {
                submitting = true
                error = null
                repository.createMaintenanceRecord(
                    id,
                    title.trim(),
                    description.trim().ifBlank { null },
                ).fold(
                    onSuccess = {
                        submitting = false
                        onSaved()
                        onDismiss()
                    },
                    onFailure = {
                        submitting = false
                        error = it.message
                    },
                )
            }
        },
    )
}

@Composable
private fun LivingHouseRuleSheetBody(
    momentId: String?,
    repository: GroupSliceRepository,
    onDismiss: () -> Unit,
    onSaved: () -> Unit,
    accent: SheetAccent,
) {
    var title by remember { mutableStateOf("") }
    var description by remember { mutableStateOf("") }
    var submitting by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()

    SheetHeader(
        R.drawable.ic_qa_book,
        "Add House Rule",
        "Set an agreement for the household",
        accent = accent,
    )
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Rule title")
        SheetField(title, { title = it }, "Rule title", minHeight = 42)
    }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        FieldLabel("Description")
        SheetField(description, { description = it }, "Rule details", minHeight = 72)
    }
    error?.let { Text(it, color = Color(0xFFF87171), fontSize = 12.sp, fontFamily = PlusJakartaSans) }
    PrimaryCta(
        label = if (submitting) "Saving…" else "Add Rule",
        enabled = !momentId.isNullOrBlank() && title.isNotBlank() && description.isNotBlank() && !submitting,
        loading = submitting,
        accent = accent,
        lightLabel = true,
        onClick = {
            val id = momentId ?: return@PrimaryCta
            scope.launch {
                submitting = true
                error = null
                repository.createLivingRule(id, title.trim(), description.trim()).fold(
                    onSuccess = {
                        submitting = false
                        onSaved()
                        onDismiss()
                    },
                    onFailure = {
                        submitting = false
                        error = it.message
                    },
                )
            }
        },
    )
}
